### PR TITLE
Add VertexAI Embedding Model support

### LINKED ISSUE #1013: Add support for the Google VertexAI Embedding API
The new [VertexAI Embedding API](https://cloud.google.com/vertex-ai/generative-ai/docs/embeddings) provides Text and Multilingual and Multimodal support.
Implement a corresponding Text EmbeddingModel and explore the possibilities to leverage the multimodal API.

### PR DESCRIPTION
**Pull request by tzolov**
- add new spring-ai-vertex-ai-embedding project.
 - add VertexAiTextEmbeddingModel and VertexAiMultimodalEmbeddingMode with related options configuration classes.
 - add ITs
 - add auto-configuraiton and boot starters.
 - register to BOM.
 - add documentation.
 - add multimodal embedding documentation
 - extend the Embedding metdata so that it can keep references to the source document's data, Id, mediatype

 Resolves #1013
 Related to #1009

### CODE REVIEW

File: spring-ai-docs/src/main/antora/modules/ROOT/pages/api/embeddings/vertexai-embeddings-palm2.adoc [hunk 1]
```diff
@@ -1,6 +1,12 @@
-= VertexAI Embeddings
+= Google VertexAI PaLM2 Embeddings
```
Comment (markpollack) [inline spring-ai-docs/src/main/antora/modules/ROOT/pages/api/embeddings/vertexai-embeddings-palm2.adoc:1]: is this API being decommissioned this august?

Comment (tzolov) [reply spring-ai-docs/src/main/antora/modules/ROOT/pages/api/embeddings/vertexai-embeddings-palm2.adoc:1]: It is very confusing information. 

According to this message  https://ai.google.dev/palm_docs/deprecation it should have been deprecated in August. But then mind this note:
> Note that this deprecation notice does not pertain to PaLM API support in Vertex AI.

Same here: https://ai.google.dev/palm_docs/palm
> The PaLM API is deprecated for use with Google AI services and tools (BUT NOT FOR Vertex AI).

Not sure what the "Not for Vertex AI" means?

File: spring-ai-core/src/main/java/org/springframework/ai/model/ChatModelDescription.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+package org.springframework.ai.model;
+
+/**
+ * @author Christian Tzolov
+ */
+public interface ChatModelDescription extends ModelDescription {
+
+	default int getContextLength() {
```
Comment (markpollack) [inline spring-ai-core/src/main/java/org/springframework/ai/model/ChatModelDescription.java:24]: this method isn't used anywhere, so maybe we shouldn't have it?

File: spring-ai-core/src/main/java/org/springframework/ai/model/ModelDescription.java [hunk 3]
```diff
@@ -21,7 +21,7 @@
  */
 public interface ModelDescription {
 
-	String getModelName();
+	String getTextName();
```
Comment (markpollack) [inline spring-ai-core/src/main/java/org/springframework/ai/model/ModelDescription.java:24]: will change to `getName`

File: spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/vertexai/embedding/VertexAiMultimodalEmbeddingProperties.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import org.springframework.ai.vertexai.embedding.multimodal.VertexAiMultimodalEmbeddingOptions;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * Configuration properties for Vertex AI Gemini Chat.
+ *
+ * @author Christian Tzolov
+ * @since 1.0.0
+ */
+@ConfigurationProperties(VertexAiTextEmbeddingProperties.CONFIG_PREFIX)
```
Comment (markpollack) [inline spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/vertexai/embedding/VertexAiMultimodalEmbeddingProperties.java:27]: cut-n-paste error, should be `VertexAiMultimodalEmbeddingProperties`

File: spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/vertexai/embedding/VertexAiEmbeddingConnectionProperties.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+/**
+ * Configuration properties for Vertex AI Embedding.
+ *
+ * @author Christian Tzolov
+ * @since 1.0.0
+ */
+@ConfigurationProperties(VertexAiEmbeddingConnectionProperties.CONFIG_PREFIX)
+public class VertexAiEmbeddingConnectionProperties {
+
+	public static final String CONFIG_PREFIX = "spring.ai.vertex.ai.embedding";
```
Comment (markpollack) [inline spring-ai-spring-boot-autoconfigure/src/main/java/org/springframework/ai/autoconfigure/vertexai/embedding/VertexAiEmbeddingConnectionProperties.java:30]: note, there is the same prefix on `VertexAiPalm2EmbeddingProperties` which will likely be removed as that impl uses the soon to be decomissioned palm api key.

**Comment by markpollack** [comment]
reviewed with minor changes in c70c20b79d0199aa031d53438b9b988b49473e5e

**Pull request closed without merging**